app/src/main/java/com/spendster/presentation/signUp/SignUpPresenter.java

```diff
@@ -10,31 +10,40 @@ public SignUpPresenter(SignUpView signUpView) {
     private boolean emailIsValid(String email){
         boolean check = true;
         if(email.isEmpty()){
-            signUpView.showEmailIsEmptyError();
+            signUpView.showEmailError("Email field is empty");
```

Comment:
@SimonMaha It's better to create a String variable errorMessage and assign value to it. In the last place show error on view.

Reply:
@SimonMaha it's pretty similar with many "return" in function